exec/vm.go

```diff
@@ -60,6 +60,8 @@ type VM struct {
 	compiledFuncs []compiledFunction
 
 	funcTable [256]func()
+
+	RecoverExecutionPanic bool // Set to `true` to get an error instead of panicing when execution fails, e.g. because of reaching `unreachable`
```

Comment:
Optional suggestion for how this can be phrased differently:

```Go
// RecoverExecutionPanic controls whether ExecCode method
// recovers execution panics and returns them as errors instead.
// A panic can only occur when executing an invalid VM.
RecoverExecutionPanic bool
```

I'm not sure if the last sentence is factually correct, so please update accordingly, but I think it's a good idea to explain the situations when panics would occur (i.e., it's not meant to catch bugs in the interpreter implementation).

Reply:
Will do. For reference, `panic`s do occur when the VM is valid and execute invalid software (e.g. `unreachable` is reached, which can happen as the instruction could be found in the binary flow of some WASM file)